### DISCUSSION TITLE
Add HTMX Support

### DISCUSSION THREAD

**Discussion by ryankopf** [loco-rs/loco#362] [General]
## Feature Request

HTMX is great and hugely up-and-coming in the frontend world. HTMX is a powerful tool that allows us to enrich our web applications with dynamic content, without the complexity of a full-blown JavaScript framework. This addition would streamline the development process, significantly reduce the learning curve for new developers, and enhance the overall user experience.

Key Benefits:
- Very "Railsy" - HTMX is a lot like turbo and Hotwire, except its not currently just limited to Rails. Integration will be relatively simple.
- Ease of Use: HTMX's syntax is intuitive and aligns well with our existing HTML-centric development approach. It simplifies the process of adding interactive features to web pages.
- Improved Performance: By using HTMX, we can minimize the amount of JavaScript needed, leading to faster page loads and a smoother user experience, especially on lower-powered devices.
- Enhanced Scalability: HTMX's lightweight nature and compatibility with standard HTTP methods make it a scalable solution for both small and large-scale applications.
- Seamless Integration: Given its design, HTMX can be seamlessly integrated with the existing Rust architecture, maintaining the robustness and reliability we've established.
- Reduced Complexity: It reduces the complexity of our codebase by replacing intricate JavaScript functionalities with simple HTML extensions, making it easier to maintain and debug.

**Comment by jondot** [comment]
Thanks @ryankopf 
As much as I was able to gather, htmx is indeed closest to what Rails has today with Turbo.
Would you like to submit some draft or PR that showcases how it all should work?

**Comment by ryankopf** [comment]
I've created a PR that is necessary to start on this. https://github.com/loco-rs/loco/pull/344

My PR renames "SaaS App" to "React Frontend". This is necessary so that when I go to create the HTMX frontend, it will make idiomatic sense what all the options are.

In the future, imagine generating a project will look like this:

```
What would you like to build?
> Stateless App
> API-Only App
> React Frontend
> HTMX Frontend
```

**Comment by jondot** [comment]
Thanks Ryan ❤️ , moving this to discussion (after I reviewed the PR)

**Comment by MoonKibong** [comment]
I look forward to it!!!

FYI, I found you can easily(?) implement an HTMX front end for a Loco back end even as of now.
Here's the example project I've created. => https://github.com/MoonKibong/loco_htmx_full
Sorry that the documentation is in Korean because I made this example to show my fellows the ease and power of two frameworks who are mostly non-developers. I'm not a developer either. :)
It would be great if anyone add features to the repo. For example, I couldn't find a proper way to use Cookies so that only authenticated users can access protected pages/services.

**Comment by jondot** [comment]
Happy to say Loco supports HTMX, and you can also use it through generators now for generating a full HTMX powered CRUD app https://loco.rs/casts/007-htmx/
